CMakeLists.txt

```diff
@@ -57,56 +55,28 @@ if(USE_ESSLBM)
 endif()
 
 add_library(lbmLib)
+target_compile_features(lbmLib PRIVATE cxx_std_17)
 
-set(EXTERNAL_DIR ${lbm_SOURCE_DIR}/external)
-
-set(EXTERNAL_HEADER_LIST
-    ${EXTERNAL_DIR}/tinyxml/tinyxml.h
-    ${EXTERNAL_DIR}/tinyxml/tinystr.h
-    ${EXTERNAL_DIR}/zlib/crc32.h
-    ${EXTERNAL_DIR}/zlib/deflate.h
-    ${EXTERNAL_DIR}/zlib/gzguts.h
-    ${EXTERNAL_DIR}/zlib/inffast.h
-    ${EXTERNAL_DIR}/zlib/inffixed.h
-    ${EXTERNAL_DIR}/zlib/inflate.h
-    ${EXTERNAL_DIR}/zlib/inftrees.h
-    ${EXTERNAL_DIR}/zlib/trees.h
-    ${EXTERNAL_DIR}/zlib/zconf.h
-    ${EXTERNAL_DIR}/zlib/zlib.h
-    ${EXTERNAL_DIR}/zlib/zutil.h
+set(LBM_HEADER_LIST
+    incl/unistd.h
```

Comment:
This file prevents compilation on Linux (io.h does not exist on Unix systems). Maybe move it to `incl/windows/unistd.h` and then use `if (WIN32) add_subdirectory(windows) endif()`?
You can also have multiple calls to `target_sources` such that you only need the one `if` and the header will then get added to all necessary targets in the subdirectory `CMakeLists.txt`. Same for the `target_include_directories`.